com.minres.coredsl/src/com/minres/coredsl/CoreDsl.xtext

```diff
@@ -245,11 +245,11 @@ StringLiteral: value=ENCSTRINGCONST | value=STRING ;
 //////////////////////////////////////////////////////////////////////////////
 
 enum IntegerSignedness : SIGNED='signed' | UNSIGNED='unsigned';
-enum IntegerSizeShorthand : INT='int' | CHAR='char' | SHORT='short' |  LONG='long';
+enum IntegerSizeShorthand : INT='int' | CHAR='char' | SHORT='short' | LONG='long';
 enum FloatSizeShorthand : FLOAT='float' | DOUBLE='double';
 
 enum TypeQualifier: CONST='const' | VOLATILE='volatile';
-enum StorageClassSpecifier: EXTERN='extern' | STATIC='static' | REGISTER='register';
+enum StorageClassSpecifier: EXTERN='extern' | REGISTER='register' | PARAM='param';
```

Comment:
New keyword?

Reply:
Yes, I had to add a new value to the enumeration and decided to make it available as a keyword as well. I personally would even vote to make it mandatory for isa parameters (to make it clear to the reader what is an implementation parameter and what is a runtime-modifiable value), but as of now all declarations with no storage class specifier default to `param`.

Reply:
Had an offline discussion with @eyck about this: We agree that the new storage classes (incl. the one for aliases you mention below) make sense, but we don't want to add new keywords for this, in order to stay close to C. Eyck, you noted that there's a way to have the corresponding enum variants without them being a keyword; can you point @AtomCrafty in the right direction please?

Reply:
Misunderstanding on my part, he meant extending the Xtext-generated `enum` later (in Java/Xtend), for the sake of our elaboration only.

W.r.t. the grammar, `StorageClassSpecifier` is optional. The ampersand token to mark a declarator as an `alias` is also optional. So it should be possible to parse and correctly distinguish registers, address spaces, aliases and parameters, correct?

Reply:
Changed the implementation to use a different enum for the internal representation than for the AST, which also includes values for `param`, `alias` and `local`. The following table shows how each combination of declaration location and storage class specifier is mapped to a storage class:

| ↓ Location / Keyword → | None | `register` | `extern` |
| ---- | ---- | ---- | ---- |
| **Local, no alias** | `local` | *Error* | *Error* |
| **Local, with alias** | *Error* | *Error* | *Error* |
| **Arch State, no alias** | `param` | `register` | `extern` |
| **Arch State, with alias** | `alias` | `register` | `extern` |